src/vs/workbench/browser/parts/titlebar/windowTitle.ts

```diff
@@ -85,20 +86,23 @@ export class WindowTitle extends Disposable {
 				this.titleUpdater.schedule();
 			}
 		}));
-		this.updateTitleIncludesFocusedView();
-	}
-
-	private updateTitleIncludesFocusedView(): void {
-		this.titleIncludesFocusedView = this.configurationService.getValue<string>(WindowSettingNames.title)?.match(/\${focusedView}/) !== null;
 	}
 
 	private onConfigurationChanged(event: IConfigurationChangeEvent): void {
-		if (event.affectsConfiguration(WindowSettingNames.title) || event.affectsConfiguration(WindowSettingNames.titleSeparator)) {
+		if (event.affectsConfiguration(WindowSettingNames.title)) {
 			this.updateTitleIncludesFocusedView();
+		}
+
+		if (event.affectsConfiguration(WindowSettingNames.title) || event.affectsConfiguration(WindowSettingNames.titleSeparator)) {
 			this.titleUpdater.schedule();
 		}
 	}
 
+	private updateTitleIncludesFocusedView(): void {
+		const titleTemplate = this.configurationService.getValue<unknown>(WindowSettingNames.title);
+		this.titleIncludesFocusedView = typeof titleTemplate === 'string' && titleTemplate.includes('${focusedView}');
```

Comment:
@meganrogge I was not sure why you used a regex before, when `includes` should just work?

Reply:
I recalled someone once told me `includes` is less efficient 🤔

Reply:
Not on my machine at least:

<img width="1190" alt="image" src="https://github.com/microsoft/vscode/assets/900690/f0ae75f2-284a-4b78-be0a-2aeadf74b250">